components/scream/scripts/test_all_scream.py

```diff
@@ -142,7 +172,7 @@ def generate_ctest_config(self, cmake_config, extra_configs, test):
         return result
 
     ###############################################################################
-    def generate_baselines(self, test, cleanup):
+    def generate_baselines(self, test):
```

Comment:
@bartgol , why was the cleanup flag needed?

Reply:
Uhm, I'm looking at the history of test_all_scream.py. I think I wanted to avoid some files accidentally not being rebuilt when git switched the branch back to the current branch (from baseline).

I'm not sure if it's needed. In particular, I'm not 100% sure that git, when switching from branch A to branch B, changes the timestamp of `blah.cpp` if it is identical in the two branches. However, even if it doesn't, if one of the upstream deps has changed, the build of `blah.cpp.o` should be re-triggered anyways.

tl;dr: I think I was being overly pessimistic, and there is no need to wipe the build dir when switching branches.

Reply:
Wait. What if someone uses things like globbing in some cmake file. Are we sure rebuilds will always be correctly triggered? My guess is 'yes', since we re-run cmake every time.

Reply:
It looks like the cleanup was related to data, not source files.

Reply:
Whoops, everything except data.

Reply:
OK, I think the idea here is that, if the baseline commit is the same as the HEAD commit, there's no reason to rebuild scream when running the tests.

Reply:
I think you are correct; manually re-running cmake and make should be adequate to get a correct re-build of scream without needing to wipe the binary dir.